File: java/src/main/java/org/tvarley/euler/solutions/Solution007.java
/*
 * Solution for Euler problem 007
 *
 * 10001st prime
 *
 * By listing the first six prime numbers: 2, 3, 5, 7, 11, and 13, we can see
 * that the 6th prime is 13.
 *
 * What is the 10 001st prime number?
 *
 * @author tvarley
 * @date 3/8/2026
 */
package org.tvarley.euler.solutions;

import org.tvarley.euler.Solution;
import org.tvarley.euler.util.Prime;

public class Solution007 implements Solution {
  public String solve() {
    int count = 0;
    int candidate = 2;
    while (count < 10001) {
      if (Prime.isPrime(candidate)) {
        count++;
      }
      candidate++;
    }
    return Integer.toString(candidate - 1);
  }
}

File: javascript/src/euler/solution007.js
// https://projecteuler.net/problem=7

// By listing the first six prime numbers: 2, 3, 5, 7, 11, and 13,
// we can see that the 6th prime is 13.
// What is the 10 001st prime number?
//
// Answer: 104743

function isPrime(n) {
  if (n <= 1) return false;
  if (n <= 3) return true;
  if (n % 2 === 0 || n % 3 === 0) return false;

  for (let i = 5; i * i <= n; i += 6) {
    if (n % i === 0 || n % (i + 2) === 0) return false;
  }
  return true;
}

module.exports = {
  answer: () => {
    let count = 0;
    let num = 1;

    while (count < 10001) {
      num++;
      if (isPrime(num)) {
        count++;
      }
    }
    return num;
  }
};